tests/fail/both_borrows/retag_data_race_write.stack.stderr

```diff
@@ -1,14 +1,15 @@
-error: Undefined Behavior: Data race detected between (1) non-atomic write on thread `unnamed-ID` and (2) non-atomic write on thread `unnamed-ID` at ALLOC. (2) just happened here
+error: Undefined Behavior: Data race detected between (1) retag write on thread `unnamed-ID` and (2) non-atomic write on thread `unnamed-ID` at ALLOC. (2) just happened here
   --> $DIR/retag_data_race_write.rs:LL:CC
    |
 LL |         *p = 5;
-   |         ^^^^^^ Data race detected between (1) non-atomic write on thread `unnamed-ID` and (2) non-atomic write on thread `unnamed-ID` at ALLOC. (2) just happened here
+   |         ^^^^^^ Data race detected between (1) retag write on thread `unnamed-ID` and (2) non-atomic write on thread `unnamed-ID` at ALLOC. (2) just happened here
    |
 help: and (1) occurred earlier here
   --> $DIR/retag_data_race_write.rs:LL:CC
    |
 LL |         let _r = &mut *p;
    |                  ^^^^^^^
+   = help: retags permit optimizations that insert speculative (or loop-hoisted) reads/writes. Therefore from the perspective of data races, a retag has the same implications as a read/write.
```

Comment:
One possible issue here is that the user probably won't know what a "retag" is. For aliasing model messages at least there is context implying it's probably an aliasing thing, but here this is a data race message and people won't expect aliasing model terms.

Not sure what to do about that. We could either use a less precise but more common term, such as "reborrow", or we have to say at least a few words about what a retag is.

Reply:
I'd probably add a short explanation of what a retag is. When would they not correspond to a reborrow?

Reply:
Retags happen (almost) every time a value of reference is assigned. So e.g. if `x` has type `(&i32, &i32)` then passing that to a function retags both references. If `y` has type `&(&i32, &i32)` then `let x = *y;` also retags both references in the pair.

Reply:
Maybe something like `retags happens due to explicit or implicit borrowing`?

Does the initial borrow also result in a retag? Presumably those could be involved in data races too.

Reply:
Yes, every execution of `&`/`&mut` expressions results in a retag.

Reply:
So... maybe you can just add another line which says something like
```
help: retags occur on all (re)borrows and as well as when references are copied/moved
```